pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go

```diff
@@ -403,6 +403,104 @@ var _ = Describe("VMI Stats Collector", func() {
 			Expect(metrics[1].Labels).To(Equal([]string{"testNode", "test-ns", "testvmi", "networkA", "", "InternalIP"}))
 		})
 	})
+
+	Context("VMI vNIC info", func() {
+		It("should collect kubevirt_vmi_vnic_info metric with correct labels", func() {
+			vmi := &k6tv1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "test-ns",
+					Name:      "test-vmi",
+				},
+				Spec: k6tv1.VirtualMachineInstanceSpec{
+					Domain: k6tv1.DomainSpec{
+						Devices: k6tv1.Devices{
+							Interfaces: []k6tv1.Interface{
+								{
+									Name: "iface1",
+									InterfaceBindingMethod: k6tv1.InterfaceBindingMethod{
+										Bridge: &k6tv1.InterfaceBridge{},
+									},
+								},
+								{
+									Name: "iface2",
+									InterfaceBindingMethod: k6tv1.InterfaceBindingMethod{
+										Masquerade: &k6tv1.InterfaceMasquerade{},
+									},
+								},
+								{
+									Name: "iface3",
+									InterfaceBindingMethod: k6tv1.InterfaceBindingMethod{
+										SRIOV: &k6tv1.InterfaceSRIOV{},
+									},
+								},
+								{
+									Name:    "iface4",
+									Binding: &k6tv1.PluginBinding{Name: "custom-plugin"},
+								},
+							},
+						},
+					},
+					Networks: []k6tv1.Network{
+						{
+							Name:          "iface1",
+							NetworkSource: k6tv1.NetworkSource{Pod: &k6tv1.PodNetwork{}},
+						},
+						{
+							Name:          "iface2",
+							NetworkSource: k6tv1.NetworkSource{Pod: &k6tv1.PodNetwork{}},
+						},
+						{
+							Name:          "iface3",
+							NetworkSource: k6tv1.NetworkSource{Multus: &k6tv1.MultusNetwork{NetworkName: "multus-net"}},
+						},
+						{
+							Name:          "iface4",
+							NetworkSource: k6tv1.NetworkSource{Multus: &k6tv1.MultusNetwork{NetworkName: "custom-net"}},
+						},
+					},
+				},
+			}
+
+			metrics := CollectVmisVnicInfo(vmi)
+			Expect(metrics).To(HaveLen(4))
+
+			Expect(metrics[0].Labels).To(Equal([]string{"test-vmi", "test-ns", "iface1", "bridge", "pod networking", "<none>"}))
```

Comment:
@avlitman Why do you report "<none>" ? 
I believe that if it is irrelevant, we should not report a value for this label.

Reply:
but the label will still appear just with empty "". I mean I don't know a way to report in the same metrics records with less labels

Reply:
yes, it can be empty.